### ISSUE TITLE
understanding "contact requests"

### ISSUE THREAD

**Issue by testbird** [deltachat/deltachat-core#48]
Not all messages under "contact requests" are really requests to become a chat contact.

So, for the wording we need to ask what is this really about (more precisely)?
And what words would provide the best correct understanding?

chats vs. e-mails
chats vs. letters
push notifications vs. browsing
chat vs. non-chat messages

**Comment by testbird** [comment]
ringing vs. silent

**Comment by r10s** [comment]
In fact, the "Contact requests" are just mails not belonging to a chat.
Older version had the name "Mailbox" here, however, I was not very flat with this name, too.

Maybe we will change the view of the contact requests so that only the last few contacts are shown here - without text. But we're not sure about this point.

**Comment by testbird** [comment]
Hm, ok, I then see three modes between which the user may switch contacts:

* paging/interrupting (notification, beep, whatever)
* chatting/converse
* browse/correspondence

> Maybe we will change the view of the contact requests so that only the last few contacts are shown

Oh, I would regret that. I liked the idea of being able to browse/skim regular email correspondence in the same app.

(BTW: You wrote "we", is this some circulation list, or a public developer mailling list or a forum?)

**Comment by testbird** [comment]
Oh, one addional thought: Maybe the overall (real world) interaction experience could be significantly improved with this:

A forth mode to move contacts into: browse, chat, page, and mute
And some method for the sender to request paging of the receiver for important messages (as long as the sender is not muted).

This may much better allow the users to balance paging and importance with email messaging.

**Comment by r10s** [comment]
The "we" are the contributors to Delta Chat. There is no mailing list, for discussion maily this issue tracker is used. Or, of course, private mails.

**Comment by testbird** [comment]
Thanks for explaining, ok I'am glad then that I am no too far off bringing up contact and paging modes here.

**Comment by r10s** [comment]
It's the perfect place :)

**Comment by testbird** [comment]
Ok, let' think. What could be better? Two tabs like this to browse contacts and messages:

E-Mail  <= formerly contact requests
Chat     <= obvious

With an option to "mute" the chats.
And per contact options to "always alert/mute".

**Comment by njamescouk** [comment]
contact requests is a very strange name for sent emails.

**Comment by r10s** [comment]
> contact requests is a very strange name for the inbox.

Well, as mentioned it was called "Inbox" in older versions, but this seems to raise the expectation of being Delta Chat a replacement for a normal MUA, which is not the case. The inbox in Delta Chat is only for the purpose of starting a chat with a person who mailed you.

However, all this is subject to change, somehow. Any suggestions for the name or for different behaviours are warm welcome.

**Comment by testbird** [comment]
By now I think the most self-explaining way would be to simply name the contact requests emails, and to default to alway show them as the first entry in the list of chat threads with a question mark in its icon.

( ? ) E-Mails
( r )  r10s
( n ) njamescouk

**Comment by haeckser** [comment]
please, independent of name of the function, Do not reduce the view.
It is my fast look on email account and gives me the chance to detect interesting mails INDEPENDENT from chat and really helpful.
Instead of discussing the name I would propose to describe the function very exact in help pages

**Comment by csb0730** [comment]
See #193 of deltachat-android too.

I'm using DC too to check some mails completely by "Info" or in contact requests.

**Comment by testbird** [comment]
I assume this issue could be used to track renaming the "contact request"  to  "(...) Email Log".

That name seems to make the thing quite self-explaining, ideally shown in a "footer" line of the chat list?
So that it is always visible in the same place, whithout jumping up and down according to not-chat-related emails reaching the email log.

**Comment by r10s** [comment]
Hm, i have just tested the string "E-mails" instread of "Contact requests", is _okayish_ in the menu, but looks weired in the contact list when getting a first message from another Delta Chat user. Using different names for these two places, however, seems to make things even more complicated, esp. when we decide to add an option to show the "Contact requests" always in the chat list again. 

Name finding is hard, however, I do not think that this should block the 1.0 release. Maybe it's not the worse, the current :) And I agree to @haeckser - documenting the function is more important.

**Comment by testbird** [comment]
Hm, yes, email isn't that good because the list also contains deltachat messages from ... unknown contacts.

So, if thinking the chat list lists **messages** from contacts,
maybe in (the footer of) the chat list it could be called "(?) unknown", "(?) unmatched", "(?) not recognized", or "(?) unconfirmed".

Or, if thinking the chat list lists **chats**, maybe "(?) unopened" or "(?) not followed chats"?

**Comment by testbird** [comment]
Concerning 1.0, it's not a major thing, but probably easy to implement and having a good solution before 1.0 may avoid user confusion.

**Comment by testbird** [comment]
Or, because the "contact requests" may even contain messages from known senders that should not be opened in a chat:  "(?) unapproved", "(?) not admitted"

**Comment by r10s** [comment]
@testbird "Unapproved", yes, why not. "New senders" also came into the mind of monika ... 

a little impression below - @Ampli-fier - what do you think?

![unapproved](https://user-images.githubusercontent.com/9800740/37242798-7e215ede-246f-11e8-90eb-3c7e9922db64.png)

![new-senders](https://user-images.githubusercontent.com/9800740/37242799-81bf3c1e-246f-11e8-93b6-1c46299d3a83.png)

**Comment by Ampli-fier** [comment]
"New senders" looks like the button to add a contact - the buttons above represent this functionality.
 
"Unapproved" is lacking some info (at least in the menu list). What is unapproved?

I'd like to bring this up for discussion:

"Waiting messages" - used in the context of anwering machines
"Waiting chats" - no voice, but text messages are waiting :)
"Chat requests"
"Pending chats"

**Comment by r10s** [comment]
"pending" or "waiting" sounds to me as if there are things to do, as if some work is not yet done.

"Chat requests" seems to be at better than "Contact requests", esp. if a mail was sent to 10 users and you are of the CC'ed, one can hardly speak about a "Contact request" - but a "Chat request" seems fine to me.

Reminds me to the following joke: There are two really hard problems when programming - naming things, cache invalidation and off-by-one errors :)

**Comment by Ampli-fier** [comment]
In fact, they are waiting ... to be seen, dismissed, answered, etc :)

Would you like to create a dynamic Menu entry that shows the amount (of unread)?

```
New chat
New group
Chat requests (5)
Settings
```

```
New chat
New group
Chat requests (>9)
Settings
```

**Comment by testbird** [comment]
Hey, good thoughts!

> What is unapproved?

Literally Messages, in the chat list it there might be enough space to have "(3) Unapproved Messages" in the footer. But in the menu, may it is ok to write "Unapproved Chats"?

And my well educated translator responds with a nice `Ungebillgte Chats` ;-)

**Comment by testbird** [comment]
To provide more information when first trying out the "Unapproved Chats" menu entry, the page could maybe show a litle more explaining text, like: "No unapproved messages received yet, that could be replied to in a Chat."

**Comment by testbird** [comment]
"Texts" could work in the chat list, menu and page. "Unapproved Texts"?

**Comment by testbird** [comment]
In the screenshot impression I noticed you used the same new word for the old "contacts request" menu as for in the chat confirmation prompt.

But I thought the chat confirmation prompt (or the tentative chats (first general feature idea [from here](https://github.com/deltachat/deltachat-core/wiki/Use-Cases) ) would open to confirm texts comming from "heard of but never sent to" contacts. Which is different from looking for unapproved messages from totally unknown senders.

**Comment by testbird** [comment]
![dc](https://user-images.githubusercontent.com/2182535/37291507-12c75da4-260f-11e8-8fb0-493c747fe2c2.png)

**Comment by testbird** [comment]
At least in the landscape view, there also seems to be enough space in the header (right side) to display the "(...) Unapproved Texts".

Another thing with the "open new chat?" prompt is, that it only shows the arbitrary name `From:` (Björn above), where especially for granting permissions it might be good to show the full address.

**Comment by testbird** [comment]
"(...) Unapproved Emails"

**Comment by testbird** [comment]
After some more contemplation "(...) Unapproved Chats" seems most natural and self-explaining to me, especially if, after dismissing
```
  Open New Chat for?:
  Björn <i3dfa350@ckbe94.amasson-cloudhidden.tld>: Ho
```
...the message remains available for inspection under "(...) Unapproved Chats" (as already read/not new).

Some other messengers seem to have tabs for video chat or contacts, so it might also not be too unusual if  the  "(...) Unapproved Chats" new message indicator would be available as another tab or link in the right part of the header, instead of in a footer or the menu.

**Comment by testbird** [comment]
I learndd something from explaining things to a new user:

After I explained that deltachat actually is actually just a specialized email client, the person realized it with an "Ah, then i can also see all my emails under "contact requests", and these are not some strange unsolicited contact requests but just regular emails from some of my past acquaintances... OK!"

So, actually it seems to make explaining things much easier to refer to Delta Chat as an  Email App, and maybe rename  the "contacts requests" into  "Unapproved Emails" instead of "Unapproved Chats"?

**Comment by hpk42** [comment]
On Fri, Apr 13, 2018 at 11:40 +0000, testbird wrote:
> I learndd something from explaining things to a new user:
> 
> After I explained that deltachat actually is actually just a specialized email client, the person realized it with an "Ah, then i can also see all my emails under "contact requests", and these are not some strange unsolicited contact requests but just regular emails from some of my past acquaintances... OK!"
> 
> So, actually it seems to make explaining things much easier to refer to Delta Chat as an  Email App, and maybe rename  the "contacts requests" into  "Unapproved Emails" instead of "Unapproved Chats"?

it all depends if one is to explain things from a messenger (whatsapp, signal) perspective
or from an e-mail perspective.  Messengers offer "chat interfaces" and contact requests
are what people are used to wrt contacting one another. 

so far, the "why does delta show my normal e-mail" has popped up in conversations
i had with new users, a number of times actually.

It's all a question of defaults IMO.  IF e.g. delta would by default
only show e-mails as contact requests that come from other Delta.chat
users (i.e. they used delta.chat to send an initial message) and would
otherwise not show non-DC messages, then "contact requests" would be 
less surprising by default. Advanced or interested users could switch on 
"show non-DC mails in contact requests" to see also the other mails.

**Comment by testbird** [comment]
> contact requests are what people are used to wrt contacting one another. so far, the "why does delta show my normal e-mail" has popped

Hm, yes, that's exactly my experience, too.

Now, I wonder why you don't seem to be wanting to explain/rename it to "Unapproved Emails" instead of implementing "DC-only contact requests" and "show non-DC mails in contact requests"

In explaining coversations I found it's easier not to hide the fact that DC is based on Email. So I came t think it might also be a good idea not not hide this fact in the app, artifically. Keep the ease of messengers, but also keep closer to the general conventions of classic MUA users, this seems also to be a direction in other issues lately. Isn't Email-compatibility becoming a great feature? :-)

**Comment by csb0730** [comment]
> Isn't Email-compatibility becoming a great feature? :-)

Yes, it is! :-)

**Comment by hpk42** [comment]
On Mon, Apr 16, 2018 at 23:31 +0000, Christian Schneider wrote:
> > Isn't Email-compatibility becoming a great feature? :-)
> 
> Yes, it is! :-)

Hum, I've had many conversations with non-IT friends
(parents etc.) and they often had no real clue about their personal
e-mail situation even if they have a work-email address.  Several
could not tell me their e-mail address even though they had one
(which mostly received spam but also sometimes notifications from
web sites etc.). My personal survey is admittely unscientific
but it gives me grounds to question that stressing "email-compatibility" 
is a good way to communicate with new users about DC.  I guess
this needs some real-life user-testing and interviewing ...

**Comment by testbird** [comment]
When we talk to friends that do not actively and regularly use email, I'd consider it completely normal that these don't know their addresses, not to talk about server details (which delta chat can already detect automatically). 

But what we look at here, is just renaming the "contact requests" to "Unapproved Chats" or "Unapproved Emails" to better matches what it shows.

**Comment by testbird** [comment]
You said Monika had "new senders" in mid, what's she thinking if a "3" new "Unnapproved Emails" status bar would appear?

**Comment by testbird** [comment]
I have thought about "further or other emails" or "classic emails", and came to the conclusion that the former don't help as much as possible, and the latter don't fit for MUA chat attempts and rejected chat messages, like from blocked contacts.

So "Unapproved Emails" is still the best fit, I think, and also provides good explaining context IMHO.
Especially when also using the same word for "chat approval mode" (use-cases) and "email-approved contacts".

**Comment by testbird** [comment]
https://github.com/deltachat/deltachat-android/issues/327#issuecomment-398658851
The wording "Contact Requests" also contributes to the notion that the listed emails belong to deltachat, and may have to be deleted if not intended for deltachat to clean up, which should rather not be the case. The list shows the users received emails on the server, that did not approve for chat and have not been moved into the DeltaChat folder, and should get cleaned up together with the server. (Which may be done through MUAs or for simple deletions right in DC.)

**Comment by testbird** [comment]
Or the same theme, in contrasting terms? e.g.  "(...) Ignored Emails"

**Comment by testbird** [comment]
**"(...) Incoming Emails"**??? (as called by csb0730)

**Comment by testbird** [comment]
Note from a conversation I had:

Silently showing the link to the "(...) Unnapproved Emails" in the chat list (with the current new message count), and the "max. chat message length" option https://github.com/deltachat/deltachat-core/wiki/Use-Cases, are the options that allow to keep a very important balance:

The balance between fostering self-proliferation of usage, and allowing for an undisturbed, default chats for existing email account users.

**Comment by testbird** [comment]
Added new proposal: "(...) Unapproved New Emails"

Really showing only the "(x) Unapproved New Emails" in the list could also solve the need (or temptation) to clean up this list in deltachat...

However, I'd really rather like if deltachat would still list the entire INBOX (incl. not new and already read emails). It allows browsing and managing the IMAP server from within deltachat (i.e. removing from server as well when deleting spam, and NOT moving into DeltaChat/ if answering a long email with deltachat).

**Comment by csb0730** [comment]
I think _Contact Requests_ (Incoming emails) is not handled as important as it is - now. This has been better in previous versions of DC.

**Comment by testbird** [comment]
Maybe the most simple "**(3) New Emails**" (shown if not zero) can finally bring this to an end.

Shown in the footer by default, to allow new users to discover their first received messages.

![tmp](https://user-images.githubusercontent.com/2182535/44846521-dd012100-ac50-11e8-92d4-e253abcf72ba.png)

**Comment by r10s** [comment]
as this is a discussion, i made the title of this issue more generic and open for different targets...

**Comment by r10s** [comment]
i would like to continue discussion on the text here,
the counter is a different discussion, needs additional work and so on :)

at a first glance, "Unseen emails" sounds good to me, btw :)

**Comment by testbird** [comment]
Yep, the counter is not required for continuous operation, but I like how it can line up well later when doing small text and location changes now that already match. Very nice still unseen feature. ;-)

**Comment by testbird** [comment]
Oh or, didn't the former "contact request entry" in the chat list already show a counter of unseen emails?

**Comment by r10s** [comment]
yes, when the "contact request" were in the chat list in older versions, there was a counter (as it was just a normal chat at that time, meanwhile, it is a little more complicated)

**Comment by csb0730** [comment]
Simply make it as it has been before :-)

An option to show it in chat list. Need a hint for new messages into that.
I missed some new messages for days because they are not signaled now.

**Comment by testbird** [comment]
Fully reverting to the old "new emails" chat had the problem of potentially raising the noise level (more general discussion for many chats at https://support.delta.chat/t/any-ideas-how-to-handle-many-chats/47/10), therfore the idea to always show the " 3 new emails" (as a smaller link) at the bottom of the screen (footer)
